queue/example_connection_pool_test.go

```diff
@@ -197,12 +203,17 @@ func Example_connectionPool() {
 	}
 
 	// Switch a master instance in the pool.
-	roles := []bool{true, false}
+	roles := map[string]bool{
+		"127.0.0.1:3014": true,
+		"127.0.0.1:3015": false,
+	}
 	for {
-		err := test_helpers.SetClusterRO(servers, connOpts, roles)
+		err := test_helpers.SetClusterRO(poolDialers, connOpts, roles)
```

Comment:
```suggestion
                instances := map[Dialer]bool{
                       poolDialers["127.0.0.1:3014"]: true,
                       poolDialers["127.0.0.1:3015"]: false,
                }
		err := test_helpers.SetInstancesRO(instances, connOpts)
```

Reply:
This can be inconvenient in some cases. If we have several identical dialers with different IDs in the pool.

Reply:
Changed to accepting `makeDialer` argument.